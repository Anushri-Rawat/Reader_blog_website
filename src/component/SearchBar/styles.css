.searchBar-container {
  padding-right: 2rem;
}
.search-bar {
  outline: none;
  font-size: 100%;
  background: rgba(250, 250, 250, 1)
    url(https://static.tumblr.com/ftv85bp/MIXmud4tx/search-icon.png) no-repeat
    20px center;
  border: solid 1px #ccc;
  padding: 10px 10px 10px 44px;
  display: flex;
  align-items: center;
  width: 280px;
  border-radius: 10em;
}
.search-bar::placeholder {
  font-size: 18px;
}
.sort-container {
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  border-radius: 0.375rem;
  border: 1px solid #6c757d !important;
  color: #6c757d;
}
@media (max-width: 425px) {
  .search-bar {
    padding: 8px 8px 8px 44px;
    width: 180px;
  }
  .searchBar-container {
    padding-right: 0;
  }
}
