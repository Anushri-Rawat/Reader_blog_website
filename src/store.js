import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import posts from "./reducers";

const initialState = {
  loading: false,
  posts: JSON.parse(localStorage.getItem("posts")) || [],
};

const store = () =>
  createStore(posts, initialState, composeWithDevTools(applyMiddleware(thunk)));

export default store;
