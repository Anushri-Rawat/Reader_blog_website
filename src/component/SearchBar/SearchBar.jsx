import React, { useEffect, useMemo } from "react";
import { useDispatch } from "react-redux";
import "./styles.css";
import _ from "lodash";

const SearchBar = ({ setSearchQuery }) => {
  const dispatch = useDispatch();

  const handleChange = (e) => {
    setSearchQuery(e.target.value);
  };

  const debouncedResults = useMemo(() => {
    return _.debounce(handleChange, 500);
  }, []);

  useEffect(() => {
    return () => {
      debouncedResults.cancel();
    };
  });
  return (
    <div
      className="d-flex justify-content-between align-items-center searchBar-container"
      data-testid="search-bar"
    >
      <input
        className="search-bar"
        type="search"
        placeholder="Search"
        onChange={debouncedResults}
      />
      <select
        className="sort-container"
        onChange={(e) => {
          dispatch({ type: "SORT_POSTS", payload: e.target.value });
        }}
      >
        <option value="id">Id</option>
        <option value="title">Title</option>
        <option value="userId">User Id</option>
      </select>
    </div>
  );
};

export default SearchBar;
