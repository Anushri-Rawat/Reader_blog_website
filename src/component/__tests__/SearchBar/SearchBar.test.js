import { render as rtlRender, screen, cleanup } from "@testing-library/react";
import SearchBar from "../../SearchBar/SearchBar";
import { Provider } from "react-redux";
import store from "../../../store";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

const render = (component) =>
  rtlRender(<Provider store={store()}>{component}</Provider>);

test("should render serachbar component", () => {
  render(<SearchBar />);
  expect(screen.getByTestId("search-bar")).toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <SearchBar />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
