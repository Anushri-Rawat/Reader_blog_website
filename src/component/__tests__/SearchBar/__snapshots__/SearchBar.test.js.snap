// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="d-flex justify-content-between align-items-center searchBar-container"
  data-testid="search-bar"
>
  <input
    className="search-bar"
    onChange={[Function]}
    placeholder="Search"
    type="search"
  />
  <select
    className="sort-container"
    onChange={[Function]}
  >
    <option
      value="id"
    >
      Id
    </option>
    <option
      value="title"
    >
      Title
    </option>
    <option
      value="userId"
    >
      User Id
    </option>
  </select>
</div>
`;
